geoalchemy2/admin/dialects/sqlite.py

```diff
@@ -28,19 +44,73 @@ def load_spatialite(dbapi_conn, connection_record):
     dbapi_conn.enable_load_extension(True)
     dbapi_conn.load_extension(os.environ["SPATIALITE_LIBRARY_PATH"])
     dbapi_conn.enable_load_extension(False)
-    dbapi_conn.execute("SELECT InitSpatialMetaData();")
+    try:
+        databases = [
+            i[-1].endswith(".gpkg") for i in dbapi_conn.execute("PRAGMA database_list;").fetchall()
+        ]
+        is_GeoPkg = any(databases)
+    except Exception:
+        is_GeoPkg = False
+    if is_GeoPkg:
+        if not dbapi_conn.execute("SELECT CheckGeoPackageMetaData();").fetchone()[0]:
+            # This only works on the main database
+            dbapi_conn.execute("SELECT gpkgCreateBaseTables();")
+        dbapi_conn.execute("SELECT AutoGpkgStart();")
+        dbapi_conn.execute("SELECT EnableGpkgAmphibiousMode();")
+    else:
+        dbapi_conn.execute("SELECT InitSpatialMetaData();")
 
 
 def _get_spatialite_attrs(bind, table_name, col_name):
-    col_attributes = bind.execute(
-        text(
-            """SELECT * FROM "geometry_columns"
-           WHERE f_table_name = '{}' and f_geometry_column = '{}'
-        """.format(
-                table_name, col_name
+    if is_gpkg(bind):
+        attrs = bind.execute(
+            text(
+                """SELECT
+                    A.geometry_type_name,
+                    A.srs_id,
+                    A.z,
+                    A.m,
+                    IFNULL(B.has_index, 0) AS has_index
+                FROM gpkg_geometry_columns
+                AS A
+                LEFT JOIN (
+                    SELECT table_name, column_name, COUNT(*) AS has_index
+                    FROM gpkg_extensions
+                    WHERE table_name = '{table_name}'
+                        AND column_name = '{column_name}'
+                        AND extension_name = 'gpkg_rtree_index'
+                ) AS B
+                ON A.table_name = B.table_name AND A.column_name = B.column_name
+                WHERE A.table_name = '{table_name}' AND A.column_name = '{column_name}';
+            """.format(
+                    table_name=table_name, column_name=col_name
```

Comment:
This query here is similar to the one in OGR. Good to compare them:

https://github.com/OSGeo/gdal/blob/6e9103bd5acb1ff8da305c4e77fa30335ffa3b70/ogr/ogrsf_frmts/gpkg/ogrgeopackagedatasource.cpp#L1594

One thing I see is that OGR puts `lower()` around the table_name. There is also some recommendation in the gpkg spec:

> Implementers should be aware of the fact that SQLite table names are not case sensitive and that table names in sqlite_master and gpkg_extensions may not have the same case. When searching for table name references, it is recommended to transform table names to lower case with the lower() function. See the Abstract Test Suite for an example.

Reply:
Indeed I missed this in the gpkg specs, thanks!